extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -70,6 +72,7 @@
     BeanContainerListenerBuildItem build(
             RecorderContext recorderContext,
             AgroalRecorder recorder,
+            DeploymentClassLoaderBuildItem deploymentClassLoader,
```

Comment:
You can just set io.quarkus.deployment.annotations.BuildStep#loadsApplicationClasses to true and the TCCL will be set to this CL.